src/fs/fs.ts

```diff
@@ -364,7 +394,7 @@ export class FileSystem extends EventEmitter implements FlashDataSource {
       storageUsed: fs.getStorageUsed(),
       lines:
         this.cachedInitialProject &&
-        this.cachedInitialProject.main === currentMainFile
+        this.cachedInitialProject.files[MAIN_FILE] === btoa(currentMainFile)
```

Comment:
I'm worried about btoa with unicode (try it and see). Can we use readBytes and then the base64 code we use in storage and similar elsewhere?

Reply:
It would have to be something like this then?
`fromByteArray(new TextEncoder().encode(currentMainFile))`

`currentMainFile` is expected as a string in the ternary expression to determine the number of lines.

Reply:
Hmm. I suppose you could count the lines off the byte array, but not so concisely.

Reply:
Changes as discussed - [`6889191`](https://github.com/microbit-foundation/python-editor-next/pull/626/commits/688919105a41b5f82e2720ee850d92f342aeeafc).